content/content.go

```diff
@@ -359,10 +359,39 @@ func parseGlobalContentConfig(configPath string) (GlobalRuleConfig, error) {
 	return conf, err
 }
 
+// updateRuleContentStatus function updates a map containing results of parsing
+// all rules, external and internal ones
+func updateRuleContentStatus(ruleContentStates map[string]types.RuleContentStatus,
+	ruleType types.RuleType, name string, loaded bool, err error) {
+	// fill-in value to be used in Error attribute
+	var parsingError = types.RuleParsingError("")
+	if err != nil {
+		parsingError = types.RuleParsingError(err.Error())
+	}
+
+	// new entry to a map
+	ruleContentStatus := types.RuleContentStatus{
+		RuleType: ruleType,
+		Loaded:   loaded,
+		Error:    parsingError,
+	}
+
+	// check for a name collision
+	_, found := ruleContentStates[name]
+	if found {
+		log.Error().Str("rule name", name).Msg("Duplicate rule name found")
+	}
+
+	// update map
+	ruleContentStates[name] = ruleContentStatus
+}
+
 // parseRulesInDir function finds all rules and their content in the specified
 // directory and stores the content in the provided map.
 // This function also aggregates list of rules with improper content.
-func parseRulesInDir(dirPath string, contentMap *map[string]RuleContent, invalidRules *[]string) error {
+func parseRulesInDir(dirPath string, ruleType types.RuleType,
+	contentMap *map[string]RuleContent, invalidRules *[]string,
+	ruleContentStates map[string]types.RuleContentStatus) error {
```

Comment:
also, maybe ruleContentStatusMap is clearer than ruleContentStates?
(I'm on fire with the naming, but that's because I got lost between status and states)